doc/whats_new/v0.23.rst

```diff
@@ -203,6 +203,14 @@ Changelog
   deprecated.
   :pr:`15806` by :user:`Chiara Marmo <cmarmo>`.
 
+- |Fix| Fix support of read-only float32 array input in ``predict``,
+  ``decision_path`` and ``predict_proba`` methods of
+  :class:`tree.DecisionTreeClassifier`, :class:`tree.ExtraTreeClassifier` and
+  :class:`ensemble.GradientBoostingClassifier` as well as ``predict`` method of
+  :class:`tree.DecisionTreeRegressor`, :class:`tree.ExtraTreeRegressor`, and
+  :class:`ensemble.GradientBoostingRegressor`.
+  :pr:`16331` by :user:`Alexandre Batisse <batalex>`.
```

Comment:
The list of affected estimators is obtained in https://github.com/scikit-learn/scikit-learn/pull/16359